### PR TITLE
feat(gcp): Improve gcp performance

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Increase GCP performance x8 and check if services are actives.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/gcp/services/compute/compute_service.py [hunk 1]
```diff
@@ -21,11 +26,29 @@ def __init__(self, audit_info):
...
+
+    def __region_threading_call__(self, call):
+        threads = []
+        for region in self.regions:
+            threads.append(threading.Thread(target=call, args=(region,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
```
Comment (jfagoagas) [inline prowler/providers/gcp/services/compute/compute_service.py]: Please, move this to the new `GCPService` class.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/gcp/lib/service/service.py [hunk 2]
```diff
@@ -8,11 +11,32 @@ def __init__(self, service, audit_info, region="global", api_version="v1"):
         self.service = service.lower() if not service.islower() else service
 
         self.api_version = api_version
-        self.project_ids = audit_info.project_ids
         self.default_project_id = audit_info.default_project_id
 
         self.region = region
         self.client = generate_client(service, api_version, audit_info)
+        self.project_ids = self.__is_api_active__(audit_info.project_ids)
```
Comment (jfagoagas) [inline prowler/providers/gcp/lib/service/service.py:22]: Include a comment here to say that the project_ids will be included in the API is active, or maybe we can change the name to `self.api_active_project_ids`

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/gcp/services/compute/compute_service.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/gcp/services/compute/compute_service.py]: Done

File: prowler/providers/gcp/lib/service/service.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/gcp/lib/service/service.py:22]: Done

File: prowler/providers/gcp/lib/service/service.py [hunk 3]
```diff
@@ -6,13 +11,61 @@ def __init__(self, service, audit_info, region="global", api_version="v1"):
...
+            t.join()
+
+    def __region_threading_call__(self, call):
+        threads = []
+        for region in self.regions:
+            threads.append(threading.Thread(target=call, args=(region,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
```
Comment (jfagoagas) [inline prowler/providers/gcp/lib/service/service.py]: I think you can have just one function for this, include a new argument with the `location/zone/region` and that's it.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/gcp/lib/service/service.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/gcp/lib/service/service.py]: Done

File: prowler/providers/gcp/services/compute/compute_service.py [hunk 4]
```diff
@@ -78,60 +80,58 @@ def __get_projects__(self):
...
-                                    disks_encryption=[
-                                        (
-                                            disk["deviceName"],
-                                            True
-                                            if disk.get("diskEncryptionKey", {}).get(
-                                                "sha256"
-                                            )
-                                            else False,
+                request = self.client.instances().list(project=project_id, zone=zone)
+                http = google_auth_httplib2.AuthorizedHttp(
```
Comment (jfagoagas) [inline prowler/providers/gcp/services/compute/compute_service.py]: Can you move this to the `GCPService` class, you can create a function like `__get_AuthorizedHttp_client__`.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/gcp/services/compute/compute_service.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/providers/gcp/services/compute/compute_service.py]: Done

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**